src/System.Tuples/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -0,0 +1,1119 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <seealso cref="ValueTuple{T1, T2}"/> struct.
+        /// </summary>
+        /// <param name="item1">The value of the tuple's first component.</param>
+        /// <param name="item2">The value of the tuple's second component.</param>
+        public ValueTuple(T1 item1, T2 item2)
+        {
+            Item1 = item1;
+            Item2 = item2;
+        }
+
+        /// <summary>
+        /// Returns a value that indicates whether the current <seealso cref="ValueTuple{T1, T2}"/> instance is equal to a specified object.
+        /// </summary>
+        /// <param name="other">The object to compare with this instance.</param>
+        /// <returns><see langword="true"/> if the current instance is equal to the specified object; otherwise, <see langword="false"/>.</returns>
+        ///
+        /// <remarks>
+        /// The <paramref name="other"/> parameter is considered to be equal to the current instance under the following conditions:
+        /// <list type="bullet">
+        ///     <item><description>It is a <seealso cref="ValueTuple{T1, T2}"/> struct.</description></item>
+        ///     <item><description>Its two components are of the same types as the current instance.</description></item>
+        ///     <item><description>Its two components are equal to those of the current instance.Equality is determined by the default object equality comparer for each component.</description></item>
+        /// </list>
+        /// </remarks>
+        public override bool Equals(object other)
+        {
+            return ((IStructuralEquatable)this).Equals(other, EqualityComparer<object>.Default); ;
+        }
+
+        /// <summary>
+        /// Returns a value that indicates whether the current <seealso cref="ValueTuple{T1, T2}"/> instance is equal to a specified <seealso cref="ValueTuple{T1, T2}"/>.
+        /// </summary>
+        /// <param name="other">The object to compare with this instance.</param>
+        /// <returns><see langword="true"/> if the current instance is equal to the specified object; otherwise, <see langword="false"/>.</returns>
+        ///
+        /// <remarks>
+        /// The <paramref name="other"/> parameter is considered to be equal to the current instance under the following conditions:
+        /// <list type="bullet">
+        ///     <item><description>It is a <seealso cref="ValueTuple{T1, T2}"/> struct.</description></item>
+        ///     <item><description>Its two components are of the same types as the current instance.</description></item>
+        ///     <item><description>Its two components are equal to those of the current instance.Equality is determined by the default object equality comparer for each component.</description></item>
+        /// </list>
+        /// </remarks>
+        public bool Equals(ValueTuple<T1, T2> other)
+        {
+            return Equals(Item1, other.Item1) && Equals(Item2, other.Item2);
```

Comment:
Any reason this can't use ```EqualityComparer<T1>.Default.Equals``` and ```EqualityComparer<T2>.Default.Equals``` rather than ```Object.Equals(Object, Object)```?  The latter is going to box value types, such that if both T1 and T2 were value types, we'll get four unnecessary allocations.